docs/errors/E0450.md

```diff
@@ -0,0 +1,18 @@
+# E0450: misleading use of ',' operator in index
+
+Using comma operator within an indexing operation is misleading, as it
+can be confused with a new array definition, or with access to multiple elements
+of the container.
+
+For example:
+
+```javascript
+let a = [1, 3, 5, 7, 9, 11];
+let b = a[0, 1, 2];
+```
+Here, the value of `b` will be equal to `a[1] = 5`.
+If the intention is to create a new array `b` with the first three elements of array `a` the correct syntax would be:
+```javascript
+let a = [1, 2, 3, 4];
+let b = a[a[0], a[1], a[2]];
```

Comment:
Must fix: This line triggers E0450. CI catches this mistake: https://github.com/quick-lint/quick-lint-js/actions/runs/4289586245/jobs/7477197134

> /__w/quick-lint-js/quick-lint-js/docs/errors/E0450.md: error: expected no error in code block #2 but found errors

I think you mean this?
```suggestion
let b = [a[0], a[1], a[2]];
```

Looking at it more closely, though, I think the comma syntax is more often used for multidimensional arrays, so I suggest you change your examples here:
```javascript
let matrix = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
];
let topLeft = matrix[0, 0]; // should be: matrix[0][0]
```

Reply:
Yes, I fixed that typo, and enhanced the error recovery around IIFE as you suggested.